### PR TITLE
Beacon-chain and Validator flag cleanup

### PR DESCRIPTION
**Pull request by 0xKiwi**
**What type of PR is this?**
Cleanup

**What does this PR do? Why is it needed?**
This PR moves a lot of beacon-chain specific flags to the beacon-chain/flags package and separates them to be better organized.
Also does a similar process for the validator flags.

### CODE REVIEW

File: beacon-chain/flags/base.go [hunk 1]
```diff
@@ -8,6 +8,8 @@ import (
 )
```
Comment (terencechain) [inline beacon-chain/flags/base.go:8]: I recommend rename `base.go` to something more appropriate

File: validator/flags/base.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+		Usage: "String to include in proposed blocks",
+	}
+	// EnableWebFlag enables controlling the validator client via the Prysm web ui. This is a work in progress.
+	EnableWebFlag = &cli.BoolFlag{
+		Name:  "web",
+		Usage: "Enables the web portal for the validator client (work in progress)",
+		Value: false,
+	}
+	// SlasherRPCProviderFlag defines a slasher node RPC endpoint.
+	SlasherRPCProviderFlag = &cli.StringFlag{
```
Comment (terencechain) [inline validator/flags/base.go:79]: this is RPC

File: validator/flags/base.go [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+		Name:  "slasher-rpc-provider",
+		Usage: "Slasher node RPC provider endpoint",
+		Value: "127.0.0.1:4002",
+	}
+	// SlasherCertFlag defines a flag for the slasher node's TLS certificate.
```
Comment (terencechain) [inline validator/flags/base.go:84]: this is RPC

File: validator/flags/base.go [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+	SlasherCertFlag = &cli.StringFlag{
+		Name:  "slasher-tls-cert",
+		Usage: "Certificate for secure slasher gRPC. Pass this and the tls-key flag in order to use gRPC securely.",
+	}
+	// DisableAccountMetricsFlag disables the prometheus metrics for validator accounts, default false.
+	DisableAccountMetricsFlag = &cli.BoolFlag{
```
Comment (terencechain) [inline validator/flags/base.go:90]: this is account management

File: validator/flags/base.go [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+		Name: "disable-account-metrics",
+		Usage: "Disable prometheus metrics for validator accounts. Operators with high volumes " +
+			"of validating keys may wish to disable granular prometheus metrics as it increases " +
+			"the data cardinality.",
+	}
+	// DisablePenaltyRewardLogFlag defines the ability to not log reward/penalty information during deployment
+	DisablePenaltyRewardLogFlag = &cli.BoolFlag{
```
Comment (terencechain) [inline validator/flags/base.go:97]: this is account management

File: beacon-chain/flags/p2p.go [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+	}
+	// SubscribeToAllSubnets subsribes the beacon node to every attestation subnet.
+	SubscribeToAllSubnets = &cli.BoolFlag{
+		Name:  "subscribe-all-subnets",
+		Usage: "Subscribe to all possible attestation subnets.",
+	}
+	// EnableUPnPFlag specifies if UPnP should be enabled or not. The default value is false.
+	EnableUPnPFlag = &cli.BoolFlag{
+		Name:  "enable-upnp",
+		Usage: "Enable the service (Beacon chain or Validator) to use UPnP when possible.",
```
Comment (prestonvanloon) [inline beacon-chain/flags/p2p.go:31]: Validator does not use this flag! (I hope)

File: beacon-chain/flags/rpc.go [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+	// HTTPWeb3ProviderFlag provides an HTTP access endpoint to an ETH 1.0 RPC.
+	HTTPWeb3ProviderFlag = &cli.StringFlag{
+		Name:  "http-web3provider",
+		Usage: "A mainchain web3 provider string http endpoint",
+		Value: "",
+	}
+	// RPCHost defines the host on which the RPC server should listen.
+	RPCHost = &cli.StringFlag{
+		Name:  "rpc-host",
+		Usage: "Host on which the RPC server should listen",
```
Comment (prestonvanloon) [inline beacon-chain/flags/rpc.go:15]:
```suggestion
		Usage: "Host on which the gRPC server should listen",
```

File: beacon-chain/flags/rpc.go [hunk 8]
```diff
@@ -0,0 +1,63 @@
+package flags
+
+import "github.com/urfave/cli/v2"
+
+var (
+	// HTTPWeb3ProviderFlag provides an HTTP access endpoint to an ETH 1.0 RPC.
+	HTTPWeb3ProviderFlag = &cli.StringFlag{
+		Name:  "http-web3provider",
+		Usage: "A mainchain web3 provider string http endpoint",
```
Comment (prestonvanloon) [inline beacon-chain/flags/rpc.go:9]:
```suggestion
		Usage: "An eth1 web3 provider string http endpoint",
```

File: beacon-chain/flags/rpc.go [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+	// RPCHost defines the host on which the RPC server should listen.
+	RPCHost = &cli.StringFlag{
+		Name:  "rpc-host",
+		Usage: "Host on which the RPC server should listen",
+		Value: "127.0.0.1",
+	}
+	// RPCPort defines a beacon node RPC port to open.
+	RPCPort = &cli.IntFlag{
+		Name:  "rpc-port",
+		Usage: "RPC port exposed by a beacon node",
```
Comment (prestonvanloon) [inline beacon-chain/flags/rpc.go:21]:
```suggestion
		Usage: "gRPC port exposed by a beacon node",
```

File: beacon-chain/flags/rpc.go [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+	}
+	// DisableGRPCGateway for JSON-HTTP requests to the beacon node.
+	DisableGRPCGateway = &cli.BoolFlag{
+		Name:  "disable-grpc-gateway",
+		Usage: "Disable the gRPC gateway for JSON-HTTP requests",
+	}
+	// GRPCGatewayHost specifies a gRPC gateway host for Prysm.
+	GRPCGatewayHost = &cli.StringFlag{
+		Name:  "grpc-gateway-host",
+		Usage: "The host on which the gateway server runs on",
```
Comment (prestonvanloon) [inline beacon-chain/flags/rpc.go:42]:
```suggestion
		Usage: "The host on which the gateway server runs on to serve JSON-HTTP requests",
```

File: beacon-chain/flags/rpc.go [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+		Value: "127.0.0.1",
+	}
+	// GRPCGatewayPort enables a gRPC gateway to be exposed for Prysm.
+	GRPCGatewayPort = &cli.IntFlag{
+		Name:  "grpc-gateway-port",
+		Usage: "Enable gRPC gateway for JSON requests",
```
Comment (prestonvanloon) [inline beacon-chain/flags/rpc.go:48]:
```suggestion
		Usage: "The port on which the gateway server runs to serve JSON-HTTP requests",
```

File: beacon-chain/flags/rpc.go [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+		Value: 3500,
+	}
+	// GPRCGatewayCorsDomain serves preflight requests when serving gRPC JSON gateway.
+	GPRCGatewayCorsDomain = &cli.StringFlag{
+		Name: "grpc-gateway-corsdomain",
+		Usage: "Comma separated list of domains from which to accept cross origin requests " +
+			"(browser enforced). This flag has no effect if not used with --grpc-gateway-port.",
```
Comment (prestonvanloon) [inline beacon-chain/flags/rpc.go:55]: This flag definitely has effect if used without changing --grpc-gateway-port.

```suggestion
			"(browser enforced).",
```

File: shared/cmd/flags.go [hunk 13]
```diff
@@ -27,135 +41,49 @@ var (
...
 	// EnableTracingFlag defines a flag to enable p2p message tracing.
 	EnableTracingFlag = &cli.BoolFlag{
 		Name:  "enable-tracing",
-		Usage: "Enable request tracing.",
-	}
-	// TracingProcessNameFlag defines a flag to specify a process name.
-	TracingProcessNameFlag = &cli.StringFlag{
-		Name:  "tracing-process-name",
-		Usage: "The name to apply to tracing tag \"process_name\"",
+		Usage: "Enable p2p message tracing.",
```
Comment (prestonvanloon) [inline shared/cmd/flags.go:56]: This isn't limited to p2p. 
```suggestion
		Usage: "Enable process tracing via OpenTracing / Jaeger exporter.",
```

File: shared/cmd/flags.go [hunk 14]
```diff
@@ -27,135 +41,49 @@ var (
...
 	}
 	// TracingEndpointFlag flag defines the http endpoint for serving traces to Jaeger.
 	TracingEndpointFlag = &cli.StringFlag{
 		Name:  "tracing-endpoint",
 		Usage: "Tracing endpoint defines where beacon chain traces are exposed to Jaeger.",
```
Comment (prestonvanloon) [inline shared/cmd/flags.go:61]:
```suggestion
		Usage: "Tracing endpoint defines where beacon chain traces are pushed to a OpenTracing/Jaeger collector.",
```

File: shared/cmd/flags.go [hunk 15]
```diff
@@ -27,135 +41,49 @@ var (
...
+	// TracingProcessNameFlag defines a flag to specify a process name.
+	TracingProcessNameFlag = &cli.StringFlag{
+		Name:  "tracing-process-name",
+		Usage: "The name to apply to tracing tag \"process_name\".",
+	}
 	// TraceSampleFractionFlag defines a flag to indicate what fraction of p2p
 	// messages are sampled for tracing.
 	TraceSampleFractionFlag = &cli.Float64Flag{
 		Name:  "trace-sample-fraction",
 		Usage: "Indicate what fraction of p2p messages are sampled for tracing.",
```
Comment (prestonvanloon) [inline shared/cmd/flags.go:73]:
```suggestion
		Usage: "Indicate what fraction of p2p messages are sampled for tracing. On scale of 0 to 1.0.",
```

File: validator/flags/account.go [hunk 16]
```diff
@@ -0,0 +1,143 @@
...
+		Name:  "show-private-keys",
+		Usage: "Display the private keys for validator accounts",
+		Value: false,
+	}
+	// SkipDepositConfirmationFlag skips the y/n confirmation prompt for sending a deposit to the deposit contract.
+	SkipDepositConfirmationFlag = &cli.BoolFlag{
+		Name:  "skip-deposit-confirmation",
+		Usage: "Skips the y/n confirmation prompt for sending a deposit to the deposit contract",
+		Value: false,
+	}
```
Comment (prestonvanloon) [inline validator/flags/account.go:142]: What is this used for? Prysm doesn't send deposit to deposit contract?

This flag is only used here: https://github.com/prysmaticlabs/prysm/blob/f75a8efc0d6584233dd6f1e4c6d40d5493355c70/validator/accounts/wallet_create.go#L128

Please fix appropriately.

**Review by prestonvanloon — CHANGES_REQUESTED** [review]

**Review by redblaze4209 — APPROVED** [review]

**Review by redblaze4209 — APPROVED** [review]

**Review by redblaze4209 — APPROVED** [review]

**Pull request closed without merging**